stdlib/src/collections/dict.mojo

```diff
@@ -644,6 +644,44 @@ struct Dict[K: KeyElement, V: CollectionElement](
             return default.value()[]
         raise "KeyError"
 
+    fn popitem(inout self) raises -> DictEntry[K, V]:
```

Comment:
Thanks for your review @gabrieldemarmiesse 
To me it is better we return DictItem or DictEntry or `Dictxxx` instead of the broad name like tuple. Unless we are not going to have DictEntry in the future but replaced them all with tuple in the future. 🤔

Reply:
I'm just mentionning this because in python, the thing returned by `popitem()` or even `items()` are tuples. See https://docs.python.org/3/library/stdtypes.html#dict.popitem .

I think that we started using DictEntry because Tuple didn't support `AnyType` yet, it worked with `AnyRegType` only. Now that Tuple doesn't have this limitation I believe that there is no need for DictEntry. But indeed we can wait for the maintainers to express their views on this. I can open an issue where it can be discussed separately. I'll consider this comment addressed. Thanks!